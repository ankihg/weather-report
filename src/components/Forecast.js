import React from 'react';
import {Container, Row, Col} from 'react-bootstrap';

export default class Forecast extends React.Component {
    render() {
        return (
            <Container>
                {
                    this.props.forecast &&
                    <Row>
                        <Col md={6}>
                            <h2>{this.props.forecast && this.props.forecast.desc}</h2>
                        </Col>

                        <Col md={6}>
                            <h3><Temp temp={this.props.forecast.temp.now} unitsSymbol={this.props.unitsSymbol}/></h3>
                            <h4><Temp temp={this.props.forecast.temp.min} unitsSymbol={this.props.unitsSymbol}/> - <Temp temp={this.props.forecast.temp.max} unitsSymbol={this.props.unitsSymbol}/></h4>
                        </Col>
                    </Row>
                }
                </Container>
        );
    }
}

class Temp extends React.Component {
    render() {
        return (<span>{this.props.temp}ยบ{this.props.unitsSymbol}</span>);
    }
}
