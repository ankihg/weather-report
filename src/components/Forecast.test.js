import React from 'react';
import Forecast from './Forecast';
import renderer from 'react-test-renderer';

test('Snapshot of Forecast component', () => {
    const component = renderer.create(
        <Forecast forecast={{desc: 'Cloudy forever', temp: {now: 55, min:54, max: 57}}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
