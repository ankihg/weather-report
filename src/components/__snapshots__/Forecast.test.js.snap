// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Forecast component 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <h2>
          Cloudy forever
        </h2>
      </div>
      <div
        className="col-md-6"
      >
        <h3>
          <span>
            55
            ยบ
          </span>
        </h3>
        <h5>
          <span>
            54
            ยบ
          </span>
           - 
          <span>
            57
            ยบ
          </span>
        </h5>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <h5>
          % humidity
        </h5>
      </div>
      <div
        className="col-md-8"
      >
        <h5>
           
           wind speed
        </h5>
      </div>
    </div>
  </div>
</div>
`;
